### PR TITLE
Unformatted Anchors can be cleared from right-click menu

### PR DESCRIPTION
**Pull request by haven1433**
A common user interaction is to notice that a table is too short and want to extend it. But doing so requires that the user manually clears any formats that are in the way of extension: such as anchors in that area from code that 'looks' like a pointer. To facilitate making this manual process easier, it should be possible to manually clear each false pointer without having to go to the pointer sources.

Basically, just let the user right-click and "Clear Format" even if the anchor has no format.

### CODE REVIEW

File: src/HexManiac.Tests/PointerModelTests.cs [hunk 1]
```diff
@@ -754,6 +754,17 @@ public class PointerModelTests {
...
 
+      [Fact]
+      public void AnchorWithNoFormatContextMenuContainsRemoveOption() {
+         var test = new BaseViewModelTestClass();
+         test.ViewPort.Edit("<000020> @20 "); // add a pointer and go to the destination
+
+         test.ViewPort.SelectionStart = new Point();
+         var contextItems = test.ViewPort.GetContextMenuItems(new Point());
+
+         contextItems.Single(item => item.Text == "Clear Format");
```
Comment (StephenKThung) [inline src/HexManiac.Tests/PointerModelTests.cs]: Should the items contain the "Clear Format" option exclusively or do you only need to check that a single "Clear Format" option is available?

File: src/HexManiac.Core/ViewModels/Visitors/ContextItemFactory.cs [hunk 2]
```diff
@@ -98,6 +98,8 @@ public class ContextItemFactory : IDataFormatVisitor {
          }
 
          anchor.OriginalFormat.Visit(this, data);
+
+         if (anchor.OriginalFormat is None) Results.AddRange(GetFormattedChildren());
```
Comment (StephenKThung) [inline src/HexManiac.Core/ViewModels/Visitors/ContextItemFactory.cs]: Not really an issue, but the "is None" threw me off for a bit since it's equivalent to "== null" in python.

File: src/HexManiac.Tests/PointerModelTests.cs (hunk 1 above)
Comment (haven1433) [reply src/HexManiac.Tests/PointerModelTests.cs]: In LINQ, the 'Single' method with an argument verifies that a single element in a collection matches the condition. So in this case, it verifies that there is exactly one element with the text "Clear Format". If there are none or more than one, it throws an exception.

I guess I should also verify that the context item does the right thing when used. I'll update the test to add that step.

File: src/HexManiac.Core/ViewModels/Visitors/ContextItemFactory.cs (hunk 2 above)
Comment (haven1433) [reply src/HexManiac.Core/ViewModels/Visitors/ContextItemFactory.cs]: yeah, I actually have a format type named `None` which represents a cell with no special formatting information: that is, 99.9% of the data. I could've named it `NoneFormat` or something like that, but it seemed redundant since it's already in the `...ViewModels.DataFormats` namespace. There are similarly `Pointer`, `Undefined`, and `Integer` formats that could be confusing. I haven't decided how or if to fix this point of confusion, but this review is not the place for such a fix.
